runtime/doc/develop.txt

```diff
@@ -266,66 +270,93 @@ API							*dev-api*
   "callback" arg as an Ex command (which can call Vimscript functions). The
   "buffer" arg could also be eliminated by treating a number "pattern" as
   a buffer number.
+- Avoid functions that depend on cursor position, current buffer, etc. Instead
+  the function should take a position parameter, buffer parameter, etc.
 
-                                                              *dev-api-naming*
-Use this format to name new RPC |API| functions:
+NAMING GUIDELINES                                       *dev-naming*
 
-    nvim_{thing}_{action}_{arbitrary-qualifiers}
+Naming is exceedingly important: the name of a thing is the primary interface
+for uses it, discusses it, searches for it, shares it... Consistent
+naming in the stdlib, API, and UI helps both users and developers discover and
+intuitively understand related concepts ("families"), and reduces cognitive
+burden. Discoverability encourages code re-use and likewise avoids redundant,
+overlapping mechanisms, which reduces code surface-area, and thereby minimizes
+bugs...
 
-If the function acts on an object then {thing} is the name of that object
-(e.g. "buf" or "win"). If the function operates in a "global" context then
-{thing} is usually omitted (but consider "namespacing" your global operations
-with a {thing} that groups functions under a common concept).
-
-Use existing common {action} names if possible:
-    - add       Append to, or insert into, a collection
-    - call      Call a function
-    - create    Create a new (non-trivial) thing
-    - del       Delete a thing (or group of things)
-    - eval      Evaluate an expression
-    - exec      Execute code
-    - fmt       Format
-    - get       Get things (often by a query)
-    - open      Open
-    - parse     Parse something into a structured form
-    - set       Set a thing (or group of things)
+Naming conventions ~
+
+In general, look for precedent when choosing a name, that is, look at existing
+(non-deprecated) functions. In particular, see below...
+
+							*dev-name-common*
+Use existing common {verb} names (actions) if possible:
+    - add:          Appends or inserts into a collection
+    - attach:       Listens to something to get events from it
+    - call:         Call a function
+    - create:       Creates a new (non-trivial) thing (TODO: rename to "def"?)
+    - del:          Deletes a thing (or group of things)
+    - eval:         Evaluates an expression
+    - exec:         Executes code
+    - fmt:          Formats
+    - get:          Gets things (often by a query)
+    - inspect:      Presents a high-level, often interactive, view
+    - open:         Opens something (a buffer, window, …)
+    - parse:        Parses something into a structured form
+    - set:          Sets a thing (or group of things)
+    - try_{verb}:   Best-effort operation, returns "nullish" on failure
```

Comment:
Maybe add `clear` as well? Came up in https://github.com/neovim/neovim/pull/21504 because diagnostic uses `reset` and in the api module there are functions like `nvim_clear_autocmds` or `nvim_buf_clear_namespace`